From: Boyuan Yang <byang@debian.org>
Date: Fri, 27 Sep 2019 00:56:33 -0400
Subject: Change ultimate fallback editor for fc

Hardly anybody knows how to use 'ed' and it is not installed
by default. Changing to 'ex' which at least is similar
and acts as the gateway to the default editor.

The original author of this patch for 93v is
Nicholas Bamber <nicholas@periapt.co.uk>. It is later refreshed
with the release of ksh 2020.0.0 branch.

Bug-Debian: https://bugs.debian.org/633524
Forwarded: not-needed
Last-Update: 2019-09-27

Signed-off-by: Boyuan Yang <byang@debian.org>
---
 config_ast.h.in          | 1 +
 features/meson.build     | 7 +++++++
 src/cmd/ksh93/data/msg.c | 2 +-
 3 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/config_ast.h.in b/config_ast.h.in
index bdc73da..3357e3c 100644
--- a/config_ast.h.in
+++ b/config_ast.h.in
@@ -66,6 +66,7 @@
 #mesondefine _pth_addr2line
 #mesondefine _pth_atos
 #mesondefine _pth_ed
+#mesondefine _pth_ex
 #mesondefine _pipe_rw
 #mesondefine _socketpair_shutdown_mode
 #mesondefine _pipe_socketpair
diff --git a/features/meson.build b/features/meson.build
index a7fbb07..3654fa4 100644
--- a/features/meson.build
+++ b/features/meson.build
@@ -246,6 +246,13 @@ else
     feature_data.set_quoted('_pth_ed', 'ed')
 endif
 
+ex = find_program('ex', required: false)
+if ex.found()
+    feature_data.set_quoted('_pth_ex', ex.path())
+else
+    feature_data.set_quoted('_pth_ex', 'ex')
+endif
+
 atos = find_program('atos', required: false)
 if atos.found()
     feature_data.set_quoted('_pth_atos', atos.path())
diff --git a/src/cmd/ksh93/data/msg.c b/src/cmd/ksh93/data/msg.c
index e7625d9..2d550c1 100644
--- a/src/cmd/ksh93/data/msg.c
+++ b/src/cmd/ksh93/data/msg.c
@@ -163,7 +163,7 @@ const char e_sptbnl[] = " \t\n";
 const char e_tolower[] = "tolower";
 const char e_toupper[] = "toupper";
 const char e_defpath[] = "/bin:/usr/bin:";
-const char e_defedit[] = _pth_ed;
+const char e_defedit[] = _pth_ex;
 const char e_unknown[] = "<command unknown>";
 const char e_traceprompt[] = "+ ";
 const char e_supprompt[] = "# ";
